osu.Game/Rulesets/UI/HitObjectContainer.cs

```diff
@@ -60,8 +70,12 @@ public class HitObjectContainer : LifetimeManagementContainer, IHitObjectContain
         internal double FutureLifetimeExtension { get; set; }
 
         private readonly Dictionary<DrawableHitObject, IBindable> startTimeMap = new Dictionary<DrawableHitObject, IBindable>();
-        private readonly Dictionary<HitObjectLifetimeEntry, DrawableHitObject> drawableMap = new Dictionary<HitObjectLifetimeEntry, DrawableHitObject>();
+
+        private readonly Dictionary<HitObjectLifetimeEntry, DrawableHitObject> aliveDrawableMap = new Dictionary<HitObjectLifetimeEntry, DrawableHitObject>();
+        private readonly Dictionary<HitObjectLifetimeEntry, DrawableHitObject> nonPooledDrawableMap = new Dictionary<HitObjectLifetimeEntry, DrawableHitObject>();
+
         private readonly LifetimeEntryManager lifetimeManager = new LifetimeEntryManager();
+        private readonly HashSet<HitObjectLifetimeEntry> allEntries = new HashSet<HitObjectLifetimeEntry>();
```

Comment:
Do you have future plans for this? Seems like it's only used for the two new properties added in this class.

Reply:
I planned to use it in a ScrollingHitObjectContainer generalization assuming the lifetime is derivable from an entry (the reverted commit), to recompute lifetime when scrolling parameter changed.